var/spack/repos/builtin/packages/cln/package.py

```diff
@@ -40,16 +41,34 @@ class Cln(AutotoolsPackage):
 
     variant("gmp", default=True, description="Enable GMP multiprecision library")
 
-    depends_on("autoconf", type="build")
-    depends_on("automake", type="build")
-    depends_on("libtool", type="build")
-    depends_on("m4", type="build")
+    # Build system
+    build_system(
+        conditional("cmake", when="@1.3:"),
+        conditional("autotools", when="@:1.2.2"),
```

Comment:
```suggestion
        conditional("autotools", when="@:1.2"),
```

minor: closes the intervals